clik/external/hal_cpu/source/cpu_hal.cpp

```diff
@@ -32,12 +35,36 @@
 
 #include "device/device_if.h"
 
+// http://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+static uint64_t round_up_pot(uint64_t v) {
+  v--;
+  v |= v >> 1;
+  v |= v >> 2;
+  v |= v >> 4;
+  v |= v >> 8;
+  v |= v >> 16;
+  v |= v >> 32;
+  v++;
+  return v;
+}
+
 cpu_hal::cpu_hal(hal::hal_device_info_t *info, std::mutex &hal_lock)
     : hal::hal_device_t(info), hal_lock(hal_lock) {
-  local_mem = (uint8_t *)malloc(local_mem_size);
+#if defined(HAL_CPU_WI_MODE)
+  local_mem = (uint8_t *)std::aligned_alloc(128, local_mem_size);
+#endif
+  if (const char *val = getenv("CA_HAL_DEBUG")) {
+    if (strcmp(val, "0") != 0) {
+      debug = true;
+    }
+  }
 }
 
-cpu_hal::~cpu_hal() { free(local_mem); }
+cpu_hal::~cpu_hal() {
+#if defined(HAL_CPU_WI_MODE)
+  std::free(local_mem);
+#endif
```

Comment:
In non-WI mode, it would be nice if the destructor could be defaulted. Would that also make it trivial?

Reply:
it could be, it was virtual before, but I don't see that we plan to derive from it, so I think it makes sense.